src/Fixer/ClassNotation/FinalInternalClassFixer.php

```diff
@@ -202,12 +260,93 @@ private function isClassCandidate(TokensAnalyzer $tokensAnalyzer, Tokens $tokens
             $tags[$tag] = true;
         }
 
-        foreach ($this->configuration['annotation_include'] as $tag => $true) {
-            if (!isset($tags[$tag])) {
-                return false; // ignore class: class-level PHPDoc does not contain all tags that has been included through configuration
+        return $this->isConfiguredAsInclude($tags);
+    }
+
+    private function isClassCandidateBasedOnAttribute(Tokens $tokens, int $index): bool
+    {
+        $attributeCandidates = [];
+
+        while ($tokens[$index]->isGivenKind(CT::T_ATTRIBUTE_CLOSE)) {
+            $attributeStartIndex = $nextMeaningfulIndex = $tokens->findBlockStart(Tokens::BLOCK_TYPE_ATTRIBUTE, $index);
+            $attributeString = '';
+
+            $attributes = [];
+            while ($nextMeaningfulIndex < $index && $nextMeaningfulIndex = $tokens->getNextMeaningfulToken($nextMeaningfulIndex)) {
+                if (!$tokens[$nextMeaningfulIndex]->isGivenKind([T_STRING, T_NS_SEPARATOR])) {
+                    if ($attributeString) {
+                        $attributeCandidates[$attributeString] = true;
+                        $attributeString = '';
+                    }
+
+                    continue;
+                }
+
+                $attributeString .= strtolower($tokens[$nextMeaningfulIndex]->getContent());
             }
+
+            $index = $tokens->getPrevNonWhitespace($attributeStartIndex);
         }
 
-        return true;
+        foreach ($attributeCandidates as $attributeCandidate => $true) {
```

Comment:
Maybe like this?

```suggestion
        foreach (array_keys($attributeCandidates) as $attributeCandidate) {
```

There's no point in defining `$true` because it's not used, we only analyse keys.